02_Development/Display/Tests/test_spl06_007.py

```diff
@@ -0,0 +1,355 @@
+
+    def test_zero_scaling_factor_raises_exception(self):
+        with self.assertRaises(ZeroDivisionError,
+                               msg="Fails to raise a ZeroDivisionError "
+                               "when initialized with a scaling factor "
+                               "equal to 0."):
+            calibrator = Calibrator((0, 0, 0, 0, 0, 0, 0, 0, 0, 0),
+                                    0,
+                                    0)
+
+    def test_returns_standard_pressure_given_actual_data(self):
+        # calibration coefficients pulled from a sensor
+        calibrator = Calibrator((199, -249,
+                                 12179, 14472, -2172, 1284, -7681, -33, -823),
+                                253952,
+                                524288)
+        # raw pressure and temperature found by solving the compensating
+        # equation for raw pressure and temperature given the above
+        # coefficients
+        self.assertAlmostEqual(calibrator.pressure(-2968390, 167393), 101000,
+                               places=-1,
+                               msg="Fails to return sea level pressure given "
+                               "data that matches that.")
+
+    def test_returns_1_5_atm_pressure_given_actual_data(self):
+        calibrator = Calibrator((199, -249,
+                                 12179, 14472, -2172, 1284, -7681, -33, -823),
+                                253952,
+                                524288)
+        self.assertAlmostEqual(calibrator.pressure(-3053970, 167393), 151500,
+                               places=-1,
+                               msg="Fails to return 1.5 atm pressure given "
+                               "data that matches that.")
+
+    def test_returns_2_atm_pressure_given_actual_data(self):
+        calibrator = Calibrator((199, -249,
+                                 12179, 14472, -2172, 1284, -7681, -33, -823),
+                                253952,
+                                524288)
+        self.assertAlmostEqual(calibrator.pressure(-3132150, 167393), 202000,
+                               places=-1,
+                               msg="Fails to return 2 atm pressure given "
+                               "data that matches that.")
+
+    def test_returns_3_atm_pressure_given_actual_data(self):
+        calibrator = Calibrator((199, -249,
+                                 12179, 14472, -2172, 1284, -7681, -33, -823),
+                                253952,
+                                524288)
+        self.assertAlmostEqual(calibrator.pressure(-3274260, 209505), 303000,
+                               places=-1,
+                               msg="Fails to return 3 atm pressure given "
+                               "data that matches that.")
+
+    def test_return_standard_temperature_given_actual_data(self):
+        calibrator = Calibrator((199, -249,
+                                 12179, 14472, -2172, 1284, -7681, -33, -823),
```

Comment:
lots of magic numbers throughout this file, would love an explanation for their source